NiL.JS/Core/GlobalContext.cs

```diff
@@ -473,9 +473,15 @@ public JSValue ProxyValue(object value)
                     else if (value is Task)
                     {
                         Task<JSValue> result;
+#if NETCORE
+                        if (value.GetType().GetTypeInfo().IsGenericType)
+                        {
+                            result = new Task<JSValue>(() => ProxyValue(value.GetType().GetProperty("Result").GetValue(value)));
+#else
                         if (value.GetType().GetTypeInfo().IsGenericType && typeof(Task<>).IsAssignableFrom(value.GetType().GetGenericTypeDefinition()))
```

Comment:
Why are you keeping old code in this case?

Reply:
I've not tested it on .net framework, so leave the old code to keep current behavior for .net framework.